src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -11178,15 +11188,9 @@ private bool ScanExplicitlyTypedLambda(Precedence precedence)
                         case SyntaxKind.InKeyword:
                         case SyntaxKind.ParamsKeyword:
                             this.EatToken();
-                            foundParameterModifier = true;
                             break;
                     }
 
-                    if (this.CurrentToken.Kind == SyntaxKind.EndOfFileToken)
```

Comment:
Why remove this bailout?

Reply:
Previously we were tracking whether we saw a `ref`, `out`, or `in` modifier to return `true` early, without scanning for a `=>`, and for this case when the partially complete parameter list was at the end of the file. To handle any of these cases now, we'd need the caller to pass in whether there was an explicit return type because the explicit return type case `T (ref x,` may be either a lambda with return type `T` or an invocation of method `T`. It didn't seem worth the complexity to keep the early return optimization, and the case of a lambda with parameters that have modifiers at the end of the file seemed unlikely.